src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/redirect/signature.js

```diff
@@ -0,0 +1,42 @@
+const OrderMgr = require('dw/order/OrderMgr');
+const Logger = require('dw/system/Logger');
+const Transaction = require('dw/system/Transaction');
+const URLUtils = require('dw/web/URLUtils');
+const Resource = require('dw/web/Resource');
+const constants = require('*/cartridge/adyenConstants/constants');
+const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
+
+function handleIncorrectSignature(order, { res, next }) {
+  Logger.getLogger('Adyen').error('Redirect signature is not correct');
+  Transaction.wrap(() => {
+    OrderMgr.failOrder(order, true);
+  });
+  res.redirect(
+    URLUtils.url(
+      'Checkout-Begin',
+      'stage',
+      'payment',
+      'paymentError',
+      Resource.msg('error.payment.not.valid', 'checkout', null),
+    ),
+  );
+  return next();
```

Comment:
Can we reuse this code in multiple controllers? I think this is used more often when a payment gets refused for example.